models.py

```diff
@@ -1,38 +1,43 @@
-from app import db
+from app import SQLAlchemy
+db = SQLAlchemy()
+
 
 class Customer(db.Model):
     __tablename__ = 'customer'
 
     id = db.Column(db.String(100), primary_key=True)
-    first_name = db.Column(db.String(50),nullable=False )
+    first_name = db.Column(db.String(50), nullable=False)
     last_name = db.Column(db.String(50))
     email = db.Column(db.String(100))
-    budgets = db.relationship('budget', backref='customer', lazy=True)
-    purchases = db.relationship('purchase', backref='customer', lazy=True)
-    backref=db.backref('purchase', lazy='joined')
+    budgets = db.relationship('Budget', backref='customer', lazy=True)
+    purchases = db.relationship('Purchase', backref='customer', lazy=True)
+    backref = db.backref('Customer', lazy='joined')
 
-    def __init__(self, name, author, published):
+    def __init__(self, id, first_name, last_name, email):
         self.id = id
-        self.author = author 
-        self.published = published
+        self.first_name = first_name
+        self.last_name = last_name
+        self.email = email
 
     def __repr__(self):
         return '<id {}>'.format(self.id)
-    
+
     def serialize(self):
         return {
-            'id': self.id, 
+            'id': self.id,
             'first_name': self.first_name,
             'last_name': self.last_name,
-            'email':self.email
-        } 
+            'email': self.email
+        }
+
 
 class Budget(db.Model):
     __tablename__ = 'budget'
 
     id = db.Column(db.Integer, primary_key=True)
-    customer_id = db.Column(db.String(100),db.ForeignKey('customer.id'),nullable=False)
-    budget = db.Column(db.Float(),nullable=False)
+    customer_id = db.Column(db.String(100), db.ForeignKey(
+        'customer.id'), nullable=False)
+    budget = db.Column(db.Float(), nullable=False)
```

Comment:
I've noticed a lot of the changes in this commit were formatting and making the code more readable; good work!

Reply:
thanks vivian!